db.py

```diff
@@ -34,45 +36,37 @@ def init_ladder_db():
     db.commit()
     db.close()
 
-def search_player_by_name(username, limit=25):
+
+def query_db(query, arguments, single=False):
     db = psycopg2.connect(dbname=dbname, user=user, password=password, host=host, port=port)
     c = db.cursor()
-    c.execute('''SELECT * FROM users WHERE bnet_id ILIKE %s OR username ILIKE %s LIMIT %s''',
-        [username + '%', username + '%', str(limit)])
-    players = c.fetchall()
+    c.execute(query, arguments)
+    if single:
+        result = c.fetchone()[0]
+    else:
+        result = c.fetchall()
     db.close()
-    return players
+    return result
+
+
+def search_player_by_name(username, limit=25):
+    return query_db('''SELECT * FROM users WHERE bnet_id ILIKE %s OR username ILIKE %s LIMIT %s''',
+                    [username + '%', username + '%', str(limit)])
```

Comment:
I don't think we need to do `str(limit)`

Reply:
Fixed